src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -204,6 +204,28 @@ export class TerminalPanel extends Panel {
 				event.stopPropagation();
 			}
 		}));
+		this._register(DOM.addDisposableListener(this._parentDomElement, DOM.EventType.DROP, (e: DragEvent) => {
+			if (e.target === this._parentDomElement || DOM.isAncestor(e.target as HTMLElement, this._parentDomElement)) {
+				if (!e.dataTransfer) {
+					return;
+				}
+				const url = e.dataTransfer.getData('URL');
+				const filePath = this._getPathFromUrl(url);
```

Comment:
Instead of doing this you should use `Uri.file(path)`, you should then be able to extract the path in the correct format from that class. https://github.com/Microsoft/vscode/blob/d02567a2e43203f20d4a1daf1ca7c489be033a4b/src/vs/base/common/uri.ts#L215